tests/operator_test.go

```diff
@@ -1216,6 +1217,9 @@ spec:
 				}),
 			table.Entry("[test_id:6308] daemonsets",
 				func() {
+					// When only have one node, updating virt-hander would get following error
+					// "Unable to mark node as unschedulable"
```

Comment:
Could you explain that a little bit? I would expect that we can update virt-handler without issues and I can't see where we are setting the node to unschedulable.

Reply:
Hi Roman, I did more investigation on it. If I only test `test_id:6308`, it would pass, but when I test both `test_id:6256` and `test_id:6308` on a one-node k8s cluster, the tests get fail. Test `test_id:6308` restarts virt-hander, but virt-hander fail to start with following error message:
```
{"Unable to mark node as unschedulable":"nodes \"$(NODE_NAME)\" not found","component":"virt-handler","level":"error","pos":"virt-handler.go:175","timestamp":"2022-04-15T02:05:23.421281Z"}
```
And I use `ps aux` to see the command of virt-hander, it shows 
```
virt-handler --port 8443 --hostname-override $(NODE_NAME) --pod-ip-address $(MY_POD_IP) --max-metric-requests 3 --console-server-port 8186 --graceful-shutdown-seconds 315 -v 2
```
NODE_NAME and MY_POD_IP does not have correct value. It seems that NODE_NAME is failed to get value from spec.nodeName. 
```
        containerEnv := []corev1.EnvVar{
                {
                        Name: "NODE_NAME",
                        ValueFrom: &corev1.EnvVarSource{
                                FieldRef: &corev1.ObjectFieldSelector{
                                        FieldPath: "spec.nodeName",
                                },
                        },
                },
                {
                        Name: "MY_POD_IP",
                        ValueFrom: &corev1.EnvVarSource{
                                FieldRef: &corev1.ObjectFieldSelector{
                                        FieldPath: "status.podIP",
                                },
                        },
                },
        }
```
Any suggestion?